monkestation/code/modules/clothing/masks/miscellaneous.dm

```diff
@@ -19,3 +19,42 @@
 								'monkestation/sound/voice/laugh/misc/big_laugh2.ogg',
 								'monkestation/sound/voice/laugh/misc/big_laugh3.ogg',
 								'monkestation/sound/voice/laugh/misc/big_laugh4.ogg')
+
+/obj/item/clothing/mask/translator
+	name = "MonkeTech AutoTranslator"
+	desc = "A small device that will translate speech."
+	icon = 'monkestation/icons/obj/clothing/masks.dmi'
+	worn_icon = 'monkestation/icons/mob/mask.dmi'
+	icon_state = "translator"
+	item_state = "translator"
+	slot_flags = ITEM_SLOT_MASK | ITEM_SLOT_NECK
+	modifies_speech = TRUE
+	var/currentlanguage = /datum/language/common
+	var/list/avail_languages = list(/datum/language/common) //basic translator has only common
+
+/obj/item/clothing/mask/translator/attack_self(mob/user)
+	. = ..()
+	currentlanguage = input("Select a new language:", "Selected language", currentlanguage) in avail_languages
+
+
+/obj/item/clothing/mask/translator/handle_speech(datum/source, list/speech_args)
+	. = ..()
+	if(!CHECK_BITFIELD(clothing_flags, VOICEBOX_DISABLED))
+		if(obj_flags & EMAGGED)
+			speech_args[SPEECH_LANGUAGE] = pick(GLOB.all_languages)
+		else
+			speech_args[SPEECH_LANGUAGE] = currentlanguage
+
+/obj/item/clothing/mask/translator/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Click while in hand to select output language.</span>"
```

Comment:
This should probably have a check for the length of the list, so it's not there for single language lists.

Reply:
decided to not include this, i believe it's too much of an out and metaknowledge if someone has, for example, one of the cult languages learned.